include/swift/Runtime/Win32.h

```diff
@@ -0,0 +1,55 @@
+//===--- Win32.h - Win32 utility functions ----------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Utility functions that are specific to the Windows port.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_RUNTIME_WIN32_H
+#define SWIFT_RUNTIME_WIN32_H
+
+#ifdef _WIN32
+
+#include "swift/shims/Visibility.h"
+
+#include <wchar.h>
+
+namespace swift {
+namespace win32 {
```

Comment:
I'm really not sure about the `win32` namespace here, this is pretty generic, and there are a few different implementations possible, some more portable than others.

Reply:
This is only required for Windows, and only because of the wide character interfaces. For those, I think we want to use the Windows API to do the conversion, because that will be consistent with other things on the platform, rather than using some generic UTF-8 to UTF-16/UTF-16 to UTF-8 conversion code.